src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -438,6 +488,8 @@ private static bool CheckConstraintsSingleType(TypeSymbol type, CheckConstraints
             Location location,
             DiagnosticBag diagnostics)
         {
+            type = (NamedTypeSymbol)type.TupleUnderlyingTypeOrSelf();
```

Comment:
Thanks for confirming. I guess we are relying on TypeSymbolExtensions.VisitType to dig though the tuple and call this method for the "Rest" tuple, and so on for the entire chain. Please confirm and add a comment regarding this here.